playwright-e2e/dsm/pages/miscellaneous-pages/user-and-permissions-page.ts

```diff
@@ -0,0 +1,307 @@
+  private readonly PAGE_TITLE: string = 'Users And Permissions';
+  private CMI_STUDY_GROUP = 'cmi';
+  private PECGS_STUDY_GROUP = 'pecgs';
+  private RGP_STUDY_GROUP = 'malab';
+  private ATCP_STUDY_GROUP = 'atcp';
+  private PRION_STUDY_GROUP = 'prion';
+  private BRUGADA_STUDY_GROUP = 'brugada';
+  private DARWIN_ARK_STUDY_GROUP = 'darwin';
+  private currentStudyPermissions: UserPermission[];
+  private readonly page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+    this.currentStudyPermissions = [];
+  }
+
+  /* Locators */
+  public getAddUserButton(): Locator {
+    return this.page.locator(`//button[contains(@class, 'addUserButton')]`);
+  }
+
+  public getStudyAdmin(email: string): Locator {
+    //return this.page.locator(`//mat-expansion-panel//mat-panel-title[contains(text(), '${email}')]`);
+    return this.page.getByText(`${email}`);
+  }
+
+  public getComparePermissionsButton(studyAdmin: Locator): Locator {
+    return studyAdmin.locator(`/following-sibling::mat-panel-description//button[@mattooltip='Compare Permissions']`);
+  }
+
+  public getEditUserButton(studyAdmin: Locator): Locator {
+    return studyAdmin.locator(`/following-sibling::mat-panel-description//button[@mattooltip='Edit User']`);
+  }
+
+  public getDeleteUserButton(studyAdmin: Locator): Locator {
+    return studyAdmin.locator(`/following-sibling::mat-panel-description//button[@mattooltip='Delete User']`);
+  }
+
+  public async getPermissionsSection(studyAdmin: Locator): Promise<Array<Locator>> {
+    return studyAdmin.locator(`//ancestor::mat-expansion-panel//section[contains(@class, 'permissions')]//app-permission-checkbox`).all();
+  }
+
+  public getSaveButton(studyAdmin: Locator): Locator {
+    return studyAdmin.locator(`//ancestor::mat-expansion-panel//button[contains(., 'Save')]`);
+  }
+
+  public getDiscardChangesButton(studyAdmin: Locator) {
+    return studyAdmin.locator(`//ancestor::mat-expansion-panel//button[contains(., 'Discard changes')]`);
+  }
+
+  /* Assertions */
+  public async assertPageTitle(): Promise<void> {
+    await expect(this.page.getByRole('heading', { name: `${this.PAGE_TITLE}` }),
+      `ERROR - ${this.PAGE_TITLE}'s page title is not displayed`).toBeVisible();
+  }
+
+  public async assertAddUserButtonDisplayed(): Promise<void> {
+    const button = this.getAddUserButton();
+    await expect(button).toBeVisible();
+  }
+
+  public async assertStudyAdminInfo(email: string, name: string, phone?: string): Promise<void> {
+    //Get the relevant study admin's section
+    const studyAdmin = this.getStudyAdmin(email);
+
+    //Verify the email is as expected
+    const studyAdminEmail = await studyAdmin.innerText();
+    expect(studyAdminEmail).toBe(email);
+
+    //Verify the name is as expected
+    const studyAdminName = await this.getStudyAdminName(name);
+    expect(studyAdminName).toBe(name);
+
+    if (phone) {
+      //If phone information is provided, verify that it is shown in the page as expected
```

Comment:
For cases like this where the `if (phone)` is already pretty clear about what's being checked, you can remove the `If phone information is provided` part of the comment, leaving something like:

```
    if (phone) {
      //verify that it is shown in the page as expected
```

This way you get clarity without repeating code in the comments.